src/content/developers/docs/oracles/index.md

```diff
@@ -37,18 +37,36 @@ To understand how an oracle works, let's play through a scenario where your smar
 
 An oracle is only as secure as its data source(s). If a dapp uses Uniswap as an oracle for its ETH/DAI price feed, it is possible for an attacker to move the price on Uniswap in order to manipulate the dapp's understanding of the current price. An example of how to combat this is [a feed system](https://developer.makerdao.com/feeds/) like the one used by MakerDAO, which collates price data from a number of external price feeds instead of just relying on a single source.
 
+
+### Architecture {#architecture}
+
+The simplest architecture of Oracles follows a simple architecture, but there are more ways than this to trigger off-chain computation. 
+
+1. Emit a log with your smart contract event
+2. An off-chain service has subscribed (usually using something like the JSON-RPC `eth_subscribe` command) to these specific logs. 
+3. The off-chain service proceeds to do some tasks as defined by the log. 
+4. The off-chain service responds with the data requested in a secondary transaction. 
+
+This is how to get data in a 1 to 1 manner, however you will want to get your data in a decentralized manner, otherwise the design of your smart contract is centralized by nature of it's data.
+
+So the next step would be to have a network of these nodes making these calls to different APIs and sources, and aggregagted on-chain.
+
+[Chainlink Off-Chain Reporting](https://blog.chain.link/off-chain-reporting-live-on-mainnet/) (Chainlink OCR) has improved on this methodology by having the off-chain oracle network communicate with each other, cryptographically signed their responses, and send only 1 transaction on-chain having aggreagated their responses off-chain. This way, fewer gas is spent, but you still get the guarantee of decentralized data since every node has signed their part of the transaction, making it unchangeable by the node sending the transaction. If the node doesn't transact, the escalation policy kicks in, and the next node sends the transaction. 
 ## Usage {#usage}
 
-### Oracles as a service {#oracles-as-a-service}
+### Oracles {#oracles}
 
-Services like Chainlink offer oracles-as-a-service for you to use. They have the infrastructure in place for you to do things like:
+Chainlink is the way to get data already decentralized and on chain, and build your own modular oracle networks to get any customized data you're looking for. Chainlink has infrastructure in place to:
```

Comment:
(missing hyphen in "on-chain")
Also, feel this sentence reads a little strange, perhaps:
`Chainlink is a way to get already decentralized, on-chain data, and build...`

Reply:
Good point.. updating